plugin/src/main/java/io/jenkins/plugins/analysis/core/steps/IssuesPublisher.java

```diff
@@ -148,6 +159,29 @@ ResultAction attachAction(final TrendChartType trendChartType) {
         return action;
     }
 
+    /**
+     * Publishes checks to platforms. Afterwards, all warnings are available in corresponding platform's UI.
+     */
+    public void publishChecks(final ResultAction action) {
+        AnalysisResult result = action.getResult();
+        IssuesStatistics totals = result.getTotals();
+
+        ChecksPublisher publisher = ChecksPublisherFactory.fromRun(run);
+        publisher.publish(new ChecksDetailsBuilder()
+                .withName(result.getId())
+                .withStatus(ChecksStatus.COMPLETED)
+                .withConclusion(extractChecksConclusion(result.getQualityGateStatus()))
+                .withOutput(new ChecksOutputBuilder()
+                        .withTitle(StringUtils.capitalize(result.getId()) + " Warnings")
+                        .withSummary(extractChecksSummary(totals))
+                        .withText(extractChecksText(totals))
+                        .withAnnotations(extractChecksAnnotations(result.getIssues()))
```

Comment:
hmm I see check annotations showing up on unchanged files do we want that?

https://github.com/XiongKezhi/codingstyle/pull/1/files

Reply:
I'm not sure how we can recognize the changed files of a PR, maybe just report new or delta issues?

Reply:
probably a separate change, should be doable though

Reply:
I think it can happen that files will get a new (or old warning) that is in a file that is not part of the PR. Simple use case: add a new checkstyle rule to the configuration. Maybe it makes sense to add it as a configurable option later? 

I think in the beginning we should start with new warnings only:

```suggestion
                        .withAnnotations(extractChecksAnnotations(result.getNewIssues()))
```

(It would make sense to have this configurable for a job later)

Reply:
On the other hand: in order to test everything it makes sense to use all warnings. Otherwise we need to change the sources to get some output...

Reply:
Probably use configuration later, so is the configuration a responsibility for consumers instead of the API itself?

BTW, what kind of tests we need here? Unit test (test the `ChecksDetails` constructed) or integration as well?

Reply:
Yes, it makes sense that consumers may offer configuration properties for the Checks API.

Reply:
For the tests: I don't think that we need integration tests. It is already kind of hard to test the API with GitHub so using the warnings plugin in these tests will make it even harder. But I think the class can be easily tested using a unit test. (And if not we can see this as a proof that the API might be changed a little bit to support testing)

Reply:
Agree on the unit-test. 

For the configurations, I'm implementing the "disable checks publishing" feature as a start, I'll push it later, then we can discuss more other configurations.

Reply:
consumers responsibility, I would go with new only for now, it's simple to introduce a spotbugs / checkstyle error for demonstration purposes